src/modules/job-manager/update.c

```diff
@@ -588,11 +592,132 @@ static void send_error_responses (struct update *update)
     }
 }
 
+static void update_expiration_from_lookup_response (struct update *update,
+                                                    flux_future_t *f)
+{
+    flux_t *h = update->ctx->h;
+    const char *R;
+    json_t *o = NULL;
+    json_error_t error;
+
+    error.text[0] = '\0';
+
+    if (flux_kvs_lookup_get (f, &R) < 0
+        || !(o = json_loads (R, 0, NULL))
+        || json_unpack_ex (o, &error, 0,
+                           "{s:{s:F}}",
+                           "execution",
+                            "expiration", &update->instance_expiration) < 0)
+        flux_log (h,
+                  LOG_ERR,
+                  "failed to unpack current instance expiration: %s",
+                  error.text);
+    json_decref (o);
+}
+
+/*  An update to resource.R has occurred. If expiration > previous
+ *  expiration, then adjust expiration of all running jobs where
+ *  no duration is set in jobspec, but the job currently has a
+ *  set expiration. This implies the expiration was set automatically
+ *  by the scheduler and needs an update.
+ */
+static void resource_update_cb (flux_future_t *f, void *arg)
+{
+    struct update *update = arg;
+    flux_t *h = update->ctx->h;
+    struct job *job;
+    zlistx_t *jobs;
+    double old_expiration = update->instance_expiration;
+
+    update_expiration_from_lookup_response (update, f);
+    flux_future_reset (f);
+
+    /*  If this is the first successful update, or the expiration was
+     *  reduced, or there are no running jobs, then there is nothing
+     *  more to do.
+     */
+    if (old_expiration == -1.
+        || (update->instance_expiration > 0.
+            && update->instance_expiration < old_expiration)
+        || update->ctx->running_jobs == 0)
+        return;
+
+    flux_log (h,
+              LOG_DEBUG,
+              "expiration updated from %.2f to %.2f",
+              old_expiration,
+              update->instance_expiration);
```

Comment:
Ok, that's a good idea. I had them at `LOG_INFO` at first but then figured someone would comment that they were really debug messages. I guess I figured incorrectly ;-)

Reply:
This was a little tricky when expiration is updated to or from unlimited. I chose to show `(+inf)` or `(-inf)` in this case, since `INFINITY` from `math.h` could be used instead of having to render a string. LMK if that isn't acceptable.